pandas/tests/dtypes/test_missing.py

```diff
@@ -545,18 +545,113 @@ def test_array_equivalent_str(dtype):
     )
 
 
-def test_array_equivalent_nested():
+@pytest.mark.parametrize(
+    "strict_nan", [pytest.param(True, marks=pytest.mark.xfail), False]
+)
+def test_array_equivalent_nested(strict_nan):
     # reached in groupby aggregations, make sure we use np.any when checking
     #  if the comparison is truthy
-    left = np.array([np.array([50, 70, 90]), np.array([20, 30, 40])], dtype=object)
```

Comment:
should we have both same-sized and mismatched-size cases?  i expect these will be non-equivalent through different paths

Reply:
I intentionally changed that here, because I assume the intent of the test was to test a numpy array of arrays. But if the arrays are the same length, the `np.array(..)` constructor actually converts this into a 2D array. 

Below in another test, I added a case for same length (and then constructed the array with a workaround first creating an empty and then filling), see `# same-length lists`.

Reply:
Added a same-length subarrays case in this test as well.